doc/ecs/create_own_content.md

```diff
@@ -0,0 +1,82 @@
+
+Jede Component-Instanz gehört zu genau einer Entitäs-Instanz. Eine Entitäts-Instanz kann einen Component-Typen nur einmal speichern.
+
+Um Components für die DSL verfügbar zu machen, siehe TBD
+
+## System erstellen
+Um eigenen Systeme zu implementieren, muss die eigene Klasse von `ECS_System` abgeleitet werden.
+
+Die Funktionalität des Systems wird in der `update`-Methode implementiert. Diese wird einmal pro Frame aufgerufen.
+
+In der `update`-Methode wird dann über das HashSet `ECS.entities`iteriert.
+Dabei sollte zuerst geprüft werden, ob die Entität das "Key"-Component enthält. 
+Das "Key"-Component ist das Component, welches dem System sagen soll, dass die Entität unter dem System fällt, die Logik des Systems also auf die Entität angewendet werden soll.
+Beispiel: Das `VelocitySystem` soll nur auf Entitäten mit dem `VelocityComponent` agieren.
+
+Beispiel aus dem `VelocitySystem`: 
+```java
+ for (Entity entity : ECS.entities) {
+            entity.getComponent(VelocityComponent.name)
+                    .ifPresent(
+                            vc -> {
+                                //YOUR CODE
+                            });
+ } 
+```
+
+Für den Fall, dass mehrere Components benötigt werden, kann die Abfrage auch verschachtelt werden. Hierbei sollte dennoch zuerst das "Key"-Component überprüft werden.
+Je nach Situation kann es sein, dass Components aufeinander aufbauen. 
+Das `VelocityComponent` ist nutzlos ohne das `PositionComponent`. 
+Im `VelocitySystem` wird daher vorausgesetzt, dass eine Entität, die das `VelocityComponent` hält, auch ein `PositionComponent` hält, ansonsten wird eine `MissingComponentException` geworfen.
+
+Beispiel aus dem `VelocitySystem`:
+```java
+ for (Entity entity : ECS.entities) {
+            entity.getComponent(VelocityComponent.name)
+                    .ifPresent(
+                            vc -> {
+                                final PositionComponent position =
+                                        (PositionComponent)
+                                                entity.getComponent(PositionComponent.name)
+                                                        .orElseThrow(
+                                                                () ->
+                                                                        new MissingComponentException(
+                                                                                "PositionComponent"));
+                                                                                    //YOUR CODE
+                            });
+```
+
+
+Systeme werden im package `ecs.systems` abgelegt und sollen den Namensschema `$WHAT_IS_THIS_SYSTEM$System`folgen.
+Jedes System muss im Konstruktor `super()` aufrufen, um die Registrierung des Systems im `SystemController` zu gewährleisten. 
```

Comment:
Ist das in Java nicht sowieso Pflicht und Usus?! Muss das extra erwähnt werden?

Reply:
Und hier wird über den Super-Konstruktor das System automatisch registriert. Wieso wird das bei den Components anders gehandhabt? Das ist ein Bruch in der Logik.